### DISCUSSION TITLE
Accessing an API that doesn't support CORs in Nuxt 3

### DISCUSSION THREAD

**Discussion by Freeptom** [nuxt/nuxt#16121] [Questions] [answered]
Hi everyone,

I'm working on a Nuxt 3 website that uses Storyblok CMS, is built with Vite and hosted on Netlify. I'm also using the default Universal Rendering Mode. 

I'd like to access the [DVLA's MOT History API](https://dvsa.github.io/mot-history-api-documentation/) however, it doesn't support Cross Origin Resource Sharing (CORS), meaning it is not possible to call it from the client side code. 

I'm unsure what to do next in order to access the api. Would adding a backend with a framework like Express be necessary in order to access it? Ideally I'd like to stick to Netlify if possible being I'm familiar with it.

Any help would be greatly appreciated!

**Comment by TheAlexLichter** [comment, marked as answer]
You can use the[ `server` folder](https://nuxt.com/docs/guide/directory-structure/server#example), create an endpoint which internally will call the API and forwards the result ☺️ 
This utilizes [nitro](https://github.com/unjs/nitro) internally.

**Reply by Freeptom** [reply]
Brilliant, thanks for such a speedy reply and pointing me to that section of the docs. That's sorted the issue out and now allows me to hit the api with no issues. Really appreciate it! 

Here's the contents of server/api/mot.ts for anyone with a similar issue:

```import axios from "axios";
export default defineEventHandler(async (event) => {
  const config = useRuntimeConfig();
  const query = getQuery(event);
  const fetchData = async () => {
    try {
      const formatReg = query.registration?.toString().toLowerCase().replace(/\s/g, "");
      const { data } = await axios.get(
        `https://beta.check-mot.service.gov.uk/trade/vehicles/mot-tests/?registration=${formatReg}`,
        {
          headers: {
            Accept: `application/json+v6`,
            "x-api-key": config.public.motLookup,
          },
        }
      );
      const str = JSON.stringify(data[0]);
      const parse = JSON.parse(str);
      return parse;
    } catch (e) {
      console.log(e);
    }
  };
  const data = await fetchData();
  return {
    data,
  };
});
```
It's then called like so:
```
  const { data } = await useFetch(() => `/api/mot/?registration=${reg}`);
```

**Reply by TheAlexLichter** [reply]
Glad I could help 🙏
